stdlib/public/core/LazyCollection.swift

```diff
@@ -20,6 +20,11 @@
 /// methods that return lazy wrappers that are themselves
 /// `LazyCollectionType`s.
 ///
+/// - Note: Many lazy collections may not offer the usual performance given by
+///   `CollectionType`, `ForwardIndexType`, or `BidirectionalIndexType`. Be
+///   aware, therefore, that general operations on `LazyCollectionType`
+///   instances may not have the documented complexity.
```

Comment:
Please remove this note; we have one instance of `LazyCollectionType` that’s like this, and one other collection that has this behavior and "is lazy” but is-not-a `LazyCollectionType`, so the note doesn’t belong here.